### PR TITLE
OPEN-7: refactor query controller + dependencies

### PR DESCRIPTION
**Pull request by MerlijnCuypers**

### CODE REVIEW

File: app/Services/OpeninghoursService.php [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+use App\Models\Service;
+use Carbon\Carbon;
+
+/**
+ *
+ */
+class OpeninghoursService
+{
+    /**
+     * erviceModel in scope
```
Comment (zero2one) [inline app/Services/OpeninghoursService.php]: ServiceModel

File: app/Services/OpeninghoursService.php [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+namespace App\Services;
+
+use App\Models\Channel;
+use App\Models\Service;
+use Carbon\Carbon;
+
+/**
+ *
+ */
+class OpeninghoursService
```
Comment (zero2one) [inline app/Services/OpeninghoursService.php]: Check out the https://murze.be/2016/10/managing-opening-hours-php/ package and see if we can use it.

File: tests/controllers/QueryContrtollerTest.php [hunk 3]
```diff
@@ -0,0 +1,291 @@
+<?php
```
Comment (zero2one) [inline tests/controllers/QueryContrtollerTest.php]: Typo in filename "tests/controllers/QueryContrtollerTest.php" > QueryControllerTest

File: tests/controllers/QueryContrtollerTest.php [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+namespace Tests\Controllers;
+
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+class QueryContrtollerTest extends \TestCase
```
Comment (zero2one) [inline tests/controllers/QueryContrtollerTest.php]: Typo in classname QueryContrtollerTest > QueryControllerTest

File: tests/controllers/QueryContrtollerTest.php [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+            [['q' => 'fullWeek', 'date' => $dateParam]],
+        ];
+    }
+
+    /**
+     * @test
+     * @group validation
+     * @dataProvider requestTypeProvider
+     **/
+    public function validate_no_serviceUri_parameter_is_a_fail($typeParams)
```
Comment (zero2one) [inline tests/controllers/QueryContrtollerTest.php]: CamelCase instead of snake_case.

File: tests/controllers/QueryContrtollerTest.php [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+        return [
+            [['q' => 'now']],
+            [['q' => 'day', 'date' => $dateParam]],
+            [['q' => 'week']],
+            [['q' => 'fullWeek', 'date' => $dateParam]],
+        ];
+    }
+
+    /**
+     * @test
```
Comment (zero2one) [inline tests/controllers/QueryContrtollerTest.php]: Start methods that perform tests always with `test`:
* Clear what the test methods are.
* No need to add `@test` annotation.

```php
public function testValidateNoServiceUriParameterIsAFail()
```

File: app/Formatters/Formatter.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Formatters;
+
+abstract class Formatter
```
[the remark this answers is not in the document]
Comment (Jelle-S) [reply app/Formatters/Formatter.php]: Best eens bekijken om dit via service provider & tagging te doen: https://laravel.com/docs/5.2/container#tagging

[the remark this answers is not in the document]
Comment (MerlijnCuypers) [reply app/Formatters/Formatter.php]: Can be handled in orther issue:
https://github.com/StadGent/laravel_site_opening-hours/issues/88

File: app/Formatters/Openinghours.php [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+     * Render data according to the given format
+     *
+     * @param  string $format to match with available formats
+     * @param  array $data   data to transform
+     * @return mixed         formatted data
+     */
+    public function render($format, $data)
+    {
+        if (!isset(self::OUTPUT_MAPPER[$format])) {
+            throw new \Exception("The given format " . $forma . " is not available in formatter " . self::class, 1);
```
Comment (Jelle-S) [inline app/Formatters/Openinghours.php]: s/$forma/$format/

File: routes/api.php [hunk 9]
```diff
@@ -11,3 +11,6 @@
 | is assigned the "api" middleware group. Enjoy building your API!
 |
 */
+
+
+Route::get('/query', 'QueryController@query');
```
Comment (Jelle-S) [inline routes/api.php]: Add a newline at the end of the file

File: tests/controllers/QueryControllerTest.php [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+     * do request according to the given format
+     */
+    protected function doRequest($type, $params = null)
+    {
+        if (is_array($params)) {
+            $params = $this->assembleParameters($params);
+        }
+
+        $url = '/api/query?' . $params;
+        //dd($url);
```
Comment (Jelle-S) [inline tests/controllers/QueryControllerTest.php]: Remove this comment.

File: database/seeds/DatabaseSeeder.php [hunk 11]
```diff
@@ -1,39 +1,44 @@
...
 {
+
     /**
      * Run the database seeds.
      *
      * @return void
      */
     public function run()
     {
+        // $this->call('xxxTableSeeder');
```
Comment (Jelle-S) [inline database/seeds/DatabaseSeeder.php]: Remove this comment.

**Pull request merged**